src/devices/Nmea0183/Sentences/ProprietaryMessage.cs

```diff
@@ -0,0 +1,110 @@
+        /// Checks this message has the correct talker id
+        /// </summary>
+        /// <param name="sentence">The sentence to check</param>
+        /// <returns>True if this input sentence matches this message type (but be careful that this message
+        /// type needs further division by arguments)</returns>
+        protected static bool Matches(TalkerSentence sentence) => Matches(sentence.Id);
+
+        /// <summary>
+        /// Creates a default message of this type
+        /// </summary>
+        protected ProprietaryMessage()
+            : base(TalkerId.Proprietary, Id, DateTimeOffset.UtcNow)
+        {
+        }
+
+        /// <summary>
+        /// Used to create a message while decoding, see base class implementation
+        /// </summary>
+        protected ProprietaryMessage(TalkerId talker, SentenceId id, DateTimeOffset time)
+            : base(talker, id, time)
+        {
+        }
+
+        /// <summary>
+        /// The hex identifier of this message type (first field of a PCDIN message)
+        /// </summary>
+        public abstract int Identifier
+        {
+            get;
+        }
+
+        /// <summary>
+        /// Decodes a value from a longer hex string (PRDIN messages contain one blob of stringly-typed hex numbers)
+        /// </summary>
+        /// <param name="input">Input string</param>
+        /// <param name="start">Start offset of required number</param>
+        /// <param name="length">Length of required number. Must be 2, 4 or 8</param>
+        /// <param name="inverseEndianess">True to inverse the endianess of the number (reverse the partial string)</param>
+        /// <param name="value">The output value</param>
+        /// <returns>True on success, false otherwise</returns>
+        /// <exception cref="ArgumentException">Length is not 2, 4 or 8</exception>
+        /// <remarks>
+        /// Other erroneous inputs don't throw an exception but return false, e.g. string shorter than expected or
+        /// value is not a hex number. This is to prevent an exception in case of a malformed message.
+        /// </remarks>
+        protected bool ReadFromHexString(string input, int start, int length, bool inverseEndianess, out int value)
+        {
+            if (length != 2 && length != 4 && length != 8)
+            {
+                throw new ArgumentException("Length must be 2, 4, or 8", nameof(length));
+            }
+
+            if (input.Length < start + length)
+            {
+                value = 0;
+                return false;
+            }
+
+            string part = input.Substring(start, length);
+
+            if (length == 4 && inverseEndianess)
+            {
+                part = part.Substring(2, 2) + part.Substring(0, 2);
+            }
+
+            if (length == 8 && inverseEndianess)
+            {
+                part = part.Substring(6, 2) + part.Substring(4, 2) + part.Substring(2, 2) + part.Substring(0, 2);
+            }
```

Comment:
could you instead TryParse first as UInt32 then if needed:

```csharp
public static UInt32 ReverseBytes(UInt32 value)
{
     return (value & 0x000000FFU) << 24 | (value & 0x0000FF00U) << 8 |
         (value & 0x00FF0000U) >> 8 | (value & 0xFF000000U) >> 24;
}
```

and cast to int in the end?

Reply:
Isn't much shorter, but probably better to understand. Done